src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -68,6 +68,10 @@ public function request(string $method, string $url, array $options = []): Respo
             $this->responseFactory->next();
         }
 
+        if (!$response instanceof ResponseInterface) {
+            throw new TransportException('The response factory callable/iterator passed to MockHttpClient must return/yield an instance of ResponseInterface.');
```

Comment:
`The response factory passed to MockHttpClient must return/yield an instance of ResponseInterface, "%s" given.`